docs/architecture_overview.md

```diff
@@ -0,0 +1,130 @@
+### BlueprintArgs
+These define the specific arguments for configuring a blueprint. Some base arguments relevant to all blueprints are provided already, and classes that override `BlueprintArgs` can add anything else they want.
+### SharedTaskState
+Some tasks require additional state that cannot be considered static configuration, such as validation functions for accurate completion of a task, or a counter for the number of tasks completed in a run so far, or any other python state that multiple different assignments may be interested in accessing or altering. This class is for the edge cases that need something more than static configuration.
+### TaskBuilder
+All tasks need to have some kind of frontend interface for workers to interact with. The `TaskBuilder` defines hooks for points in setup where this frontend is compiled and built (or located and confirmed alive), and implementations for a specific `Blueprint` should use this part of the lifecycle to set up these things.
+### TaskRunner
+Likewise, all tasks need to have some kind of backend logic that dictates the kind of information we intend to receive. For simple 'static' tasks where we give the worker some info and they give us their work, this may be just a single exchange, however more complicated tasks can put any other kind of difficult flow control here.
+### AgentState
+As different tasks will save different data, Blueprints need a way to specify what kind of data is being saved. The `AgentState` is responsible for parsing out the required information that will need to be saved during a task, viewed during review, and exported during a final compilation. It receives both the initial information given to the worker, as well as every action packet sent to or by the worker.
+
+-------------
+### Onboarding
+Blueprints also provide simple hooks for "onboarding" tasks, which can be considered as tasks that a worker must complete in order to qualify for the real thing. These hooks are added to the `TaskRunner` and `AgentState` to provide a way to set the flow for onboarding, what is saved, and how the work is evaluated. 
+
+## Architects
+`Architect`'s contain the logic surrounding deploying a server that workers will be able to access. In many cases Mephisto is being run on compute clusters that aren't directly addressable, or in different configurations between collaborators. Mephisto should be able to run a task from start to finish regardless of the server configuration a user would like to use, and `Architect`'s provide this capability.
+
+The `Architect` class is responsible for providing Mephisto with lifecycle functions for preparing, deploying, and shutting down a given server. It's also responsible with providing access to the user via a `Channel`, which defines an interface of callbacks for incoming messages and a function for outgoing messages.
```

Comment:
Might be worth adding another paragraph to the end of this as such:

"You can pick from a suite of pre-configured Architects, or create/customize your own."